icu4c/source/i18n/unicode/calendar.h

```diff
@@ -1349,6 +1349,68 @@ class U_I18N_API Calendar : public UObject {
 
+#ifndef U_HIDE_DRAFT_API
+    /**
+     * Gets the date in a leap year or not. Recalculate the current time
+     * field values if the time value has been changed by a call to * setTime().
+     * This method is semantically const, but may alter the object in memory.
+     * A "leap year" is a year that contains more days than other years (for
+     * solar or lunar calendars) or more months than other years (for lunisolar
+     * calendars like Hebrew or Chinese), as defined in the ECMAScript Temporal
+     * proposal.
+     *
+     * @param status        ICU Error Code
+     * @return       True if the date in the fields is in a Temporal proposal
+     *               defined leap year. False otherwise.
+     * @draft ICU 73
+     */
+    virtual bool inTemporalLeapYear(UErrorCode& status) const;
+
+    /**
+     * Gets The Temporal monthCode value corresponding to the month for the date.
+     * The value is a string identifier that starts with the literal grapheme
+     * "M" followed by two graphemes representing the zero-padded month number
+     * of the current month in a normal (non-leap) year and suffixed by an
+     * optional literal grapheme "L" if this is a leap month in a lunisolar
+     * calendar. The 26 possible values are "M01" .. "M13" and "M00L" .. "M12L".
+     * For the Hebrew calendar, the values are "M01" .. "M12" for non-leap year, and
+     * "M01" .. "M05", "M05L", "M06" .. "M12" for leap year.
+     * For the Chinese calendar, the values are "M01" .. "M12" for non-leap year and
+     * in leap year with another monthCode in "M01L" .. "M12L".
+     * For Coptic and Ethiopian calendar, the Temporal monthCode values for any
+     * years are "M01" to "M13". In a calendar with a leap month at the start of
+     * some years, the month code of that month would be "M00L".
+     *
+     * @param status        ICU Error Code
+     * @return       One of 26 possible strings in {"M01".."M13", "M00L".."M12L"}.
+     * @draft ICU 73
+     */
+    virtual const char* getTemporalMonthCode(UErrorCode& status) const;
+
+    /**
+     * Sets The Temporal monthCode which is a string identifier that starts
+     * with the literal grapheme "M" followed by two graphemes representing
+     * the zero-padded month number of the current month in a normal
+     * (non-leap) year and suffixed by an optional literal grapheme "L" if this
+     * is a leap month in a lunisolar calendar. The 26 possible values are
+     * "M01" .. "M13" and "M00L" .. "M12L". For Hebrew calendar, the values are
+     * "M01" .. "M12" for non-leap years, and "M01" .. "M05", "M05L", "M06"
+     * .. "M12" for leap year.
+     * For the Chinese calendar, the values are "M01" .. "M12" for non-leap year and
+     * in leap year with another monthCode in "M01L" .. "M12L".
+     * For Coptic and Ethiopian calendar, the Temporal monthCode values for any
+     * years are "M01" to "M13". In a calendar with a leap month at the start of
```

Comment:
No, 
"
For Coptic and Ethiopian calendar, the Temporal monthCode values for any years are "M01" to "M13".
"
AND
"
In a calendar with a leap month at the start of some years, the month code of that month would be "M00L".
"

NOT in Coptic and Ethiopian calendar.

Reply:
Okay.  In that case, I'd recommend replacing "In a calendar with a leap month at the start..." to "For the XX and YY calendars, which add leap months at the beginning of the year...", where XX and YY are the actual calendars that do that.  Or at least start the sentence with "for" to maintain the parallelism and clarify that you're not talking about the Coptic and Ethiopic calendars.

Reply:
This comment in this header is part of the API proposal we discussed in ICUTC. I probably need to bring that back to ICU TC to get everyone agree upon first. 
How aboout 

```
For the calendar which the leap month at the start of some years, the month code of that month would be "M00L".
```

I need to talk to Peter about this. because after reading http://www2.math.uu.se/~svante/papers/calendars/tibet.pdf it seems they prefer that leap month coded as "M01L" instead of "M00L" and just want the leap month happen before the month with the same number. @pedberg-icu

Reply:
I wonder should I just remove all the "M00L" comment all together in the API and wait till the picture is clear about Tibeten calendar first.

Reply:
see http://www2.math.uu.se/~svante/papers/calendars/tibet.pdf

"Since these correspondences are not perfect, there are occasionally two months with the same number, in which case the first of them is regarded as a leap month"

Reply:
removed that part after approved by ICU TC